src/services/quotes/source-lists/overridable-source-list.ts

```diff
@@ -30,11 +30,25 @@ export class OverridableSourceList implements IQuoteSourceList {
     return sources;
   }
 
-  getQuote(request: SourceListRequest): Promise<SourceListResponse> {
-    return this.getSourceListForId(request.sourceId).getQuote(request);
+  getQuotes(request: MultipleSourceListRequest): Promise<SourceListResponse>[] {
+    const defaultSourcesId = [];
+    const result: Promise<SourceListResponse>[] = [];
+    for (const sourceId of request.sources) {
+      if (!this.hasOverrideSourceList(sourceId)) {
+        defaultSourcesId.push(sourceId);
+      } else {
+        result.push(...this.getSourceListForId(sourceId).getQuotes({ ...request, sources: [sourceId] }));
```

Comment:
Let's try to group sources by source list, so we can call `getQuotes` with all the quotes together